src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -14,44 +14,96 @@
 from nplinker.genomics.antismash import download_and_extract_antismash_data
 from nplinker.logconfig import LogConfig
 
-
 logger = LogConfig.getLogger(__name__)
 
 NCBI_LOOKUP_URL = 'https://www.ncbi.nlm.nih.gov/assembly/?term={}'
 JGI_GENOME_LOOKUP_URL = 'https://img.jgi.doe.gov/cgi-bin/m/main.cgi?section=TaxonDetail&page=taxonDetail&taxon_oid={}'
 USER_AGENT = 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0'
-GENOME_STATUS_FILENAME = "genome_status.csv"
+GENOME_STATUS_FILENAME = "genome_status.json"
 
 
 class GenomeStatus:
-    """To represent the data status for a certain genome ID.
+    """A class to represent the status of a single genome.
 
-    Attributes:
-        original_id(str): original ID for the genome.
-        resolved_refseq_id(str): RefSeq ID generated by resolving the original genome ID.
-        attempted(str): If the attempt of resolving the genome ID has already been done. Defaults to `False`.
-        filename(str): Path to the file that stores the genome data. Defaults to "".
+    The status of genomes is tracked in a JSON file which has a name defined
+    in variable `GENOME_STATUS_FILENAME`.
     """
 
     def __init__(self,
                  original_id: str,
-                 resolved_refseq_id: str,
-                 attempted: str = 'False',
-                 filename: str = ""):
+                 resolved_refseq_id: str = "",
+                 resolve_attempted: bool = False,
+                 bgc_path: str = ""):
+        """Initialize a GenomeStatus object for the given genome.
+
+        Args:
+            original_id (str): The original ID of the genome.
+            resolved_refseq_id (str, optional): The resolved RefSeq ID of the
+                genome. Defaults to "".
+            resolve_attempted (bool, optional): A flag indicating whether an
+                attempt to resolve the RefSeq ID has been made. Defaults to False.
+            bgc_path (str, optional): The path to the downloaded BGC file for
+                the genome. Defaults to "".
+        """
         self.original_id = original_id
-        self.resolved_refseq_id = "" if resolved_refseq_id == 'None' else resolved_refseq_id
-        self.attempted = attempted == 'True'
-        self.filename = filename
+        self.resolved_refseq_id = resolved_refseq_id
```

Comment:
What now if `resolved_refseq_id` is 'None'? If we're sure this can't happen then it's fine

Reply:
I guess it's possible when http response is None. I updated the code.